packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -70,175 +74,321 @@ fn get_pkg_path(name: &str, pkg_type: PackageType) -> String {
-        let gen_ctx = ctx_import.get(2)?;
-
-        let gen_state = import_file(mv8, "./src/worker/runner/javascript/state.js", vec![
-            &hash_util,
-        ])?;
-        let fns = import_file(mv8, "./src/worker/runner/javascript/runner.js", vec![
-            &arrow,
-            &batches_prototype,
-            &experiment_ctx_prototype,
-            &sim_init_ctx_prototype,
-            &gen_ctx,
-            &gen_state,
-        ])?;
-        let fns = fns.as_array().ok_or_else(|| {
+        let experiment_ctx_prototype = ctx_import.get_index(scope, 0).ok_or_else(|| {
+            Error::V8(
+                "Couldn't get experiment_ctx_prototype from index 0 of the context.js return \
+                 values"
+                    .to_string(),
+            )
+        })?;
+        let sim_init_ctx_prototype = ctx_import.get_index(scope, 1).ok_or_else(|| {
+            Error::V8(
+                "Couldn't get sim_init_ctx_prototype from index 1 of the context.js return values"
+                    .to_string(),
+            )
+        })?;
+        let gen_ctx = ctx_import.get_index(scope, 2).ok_or_else(|| {
+            Error::V8(
+                "Couldn't get gen_ctx from index 2 of the context.js return values".to_string(),
+            )
+        })?;
+
+        let gen_state = import_file(
+            scope,
+            context,
+            "./src/worker/runner/javascript/state.js",
+            &[hash_util],
+        )?;
+        let fns = import_file(
+            scope,
+            context,
+            "./src/worker/runner/javascript/runner.js",
+            &[
+                arrow,
+                batches_prototype,
+                experiment_ctx_prototype,
+                sim_init_ctx_prototype,
+                gen_ctx,
+                gen_state,
+            ],
+        )?;
+        let fns: v8::Local<'_, v8::Array> = fns.try_into().map_err(|err| {
             Error::FileImport(
                 "./src/worker/runner/javascript/runner.js".into(),
-                "Couldn't get array (of functions) from 'runner.js'".into(),
+                format!("Couldn't get array (of functions) from 'runner.js': {err}"),
             )
         })?;
-        Ok(Self {
+
+        let start_experiment: v8::Local<'_, v8::Function> = get_function_from_array(scope, fns, 0)?;
+        let start_sim: v8::Local<'_, v8::Function> = get_function_from_array(scope, fns, 1)?;
+        let run_task: v8::Local<'_, v8::Function> = get_function_from_array(scope, fns, 2)?;
+        let ctx_batch_sync: v8::Local<'_, v8::Function> = get_function_from_array(scope, fns, 3)?;
+        let state_sync: v8::Local<'_, v8::Function> = get_function_from_array(scope, fns, 4)?;
+        let state_interim_sync: v8::Local<'_, v8::Function> =
+            get_function_from_array(scope, fns, 5)?;
+        let state_snapshot_sync: v8::Local<'_, v8::Function> =
+            get_function_from_array(scope, fns, 6)?;
```

Comment:
Please use `Function<'_>` here

Reply:
Addressed in https://github.com/hashintel/hash/pull/414/commits/f4c4f5dc5b872909afc91ff48c1042a1fc74ae2d